### ISSUE TITLE
Does nfs-ganesha with Ceph FSAL support Application Data Block (ADB)? (use of NFSv4.2 WRITE_SAME)

### ISSUE THREAD

**Issue by yician1000** [nfs-ganesha/nfs-ganesha#1021]
Hi 
Does nfs-ganesha with Ceph FSAL support Application Data Block (ADB) feature [1] in NFSv4.2? 
Following describes this ADB feature.

Application Data Block (ADB) Support

Some applications treat a file as if it were a disk and as such want
to initialize (or format) the file image.  The WRITE_SAME operation
is introduced to send this metadata to the server to allow it to write
the block contents.

[1] https://www.msystechnologies.com/blog/what-is-new-in-nfs-v4-2/

**Comment by dang** [comment]
Ganesha doesn't support WRITE_SAME.  If you try it, you'll get NFS4ERR_NOTSUPP back.

**Comment by ffilz** [comment]
Correct. Early in the 4.2 development, there was going to be a WRITE_PLUS, and one of the FSAL_GPFS developers had implemented an early version, but that development effort didn't keep up with the final path of 4.2 that switched from WRITE_PLUS to WRITE_SAME.

That said, it wouldn't necessarily be hard to implement, and it could be implemented for all FSALs (the worst is that the FSAL allocates a block of the write size and fills it with the WRITE_SAME pattern before passing to the back end filesystem).

**Comment by yician1000** [comment]
Thanks for your reply and feedback. We are using nfs-ganesha service which is integrated into Rook Ceph project and are just figuring out what NFS features ganesha provides.

Rook: https://rook.io/docs/rook/v1.10/Storage-Configuration/NFS/nfs/#prerequisites

**Comment by ffilz** [comment]
Tagging as just an enhancement. This may be something we will consider soon.
